### PR TITLE
Implement content-audit publisher SPI for Kafka

### PR DESCRIPTION
**Pull request by sswguo**
The kafka producer client itself ( not broker ) needs some [configuration](https://kafka.apache.org/documentation/#producerconfigs) except the server host/port, i put it under src/main/resources with some basic ones and default values.

### CODE REVIEW

File: content-audit/kafka-publisher/src/main/java/org/commonjava/propulsor/KafkaPublisher.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+import java.io.File;
+import java.io.FileInputStream;
+import java.util.Properties;
+
+public class KafkaPublisher
+                implements FileEventPublisher
+{
+
+    private static final String PROP_PATH = "publisher.properties";
+
```
Comment (sswguo) [inline content-audit/kafka-publisher/src/main/java/org/commonjava/propulsor/KafkaPublisher.java]: Looks i can define a config class that with annotation like @SectionName( "publisher.kafka" ) and move the config into the unified file.

File: content-audit/kafka-publisher/src/main/java/org/commonjava/propulsor/KafkaPublisher.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    public void publishFileGroupingEvent( FileGroupingEvent fileGroupingEvent ) throws FileEventPublisherException
+    {
+        doPublish( fileGroupingEvent.getEventId().toString(), writeValueAsString( fileGroupingEvent ) );
+    }
+
+    private String writeValueAsString( Object value )
+    {
+        try
+        {
+            return new ObjectMapper().writeValueAsString( value );
```
[the remark this answers is not in the document]
Comment (sswguo) [reply content-audit/kafka-publisher/src/main/java/org/commonjava/propulsor/KafkaPublisher.java]: Yeah, good point, i will go through your comments and make changes later.

File: content-audit/kafka-publisher/src/main/java/org/commonjava/propulsor/conf/KafkaPublisherConfig.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import org.commonjava.propulsor.config.section.PropertiesSectionListener;
+
+import javax.enterprise.context.ApplicationScoped;
+
+@ApplicationScoped
+@SectionName("publisher.kafka")
+public class KafkaPublisherConfig extends PropertiesSectionListener
+{
+
+    private String topic;
```
[the remark this answers is not in the document]
Comment (jdcasey) [reply content-audit/kafka-publisher/src/main/java/org/commonjava/propulsor/conf/KafkaPublisherConfig.java]: Yeah, maybe. But if we want to take that on and generalize this, we need to move the code out of a content-audit sub-module. IMO it's a pretty easy refactor.

[the remark this answers is not in the document]
Comment (sswguo) [reply content-audit/kafka-publisher/src/main/java/org/commonjava/propulsor/conf/KafkaPublisherConfig.java]: Yeah, sounds good. Forgot that it's under content-audit currently :)

**Pull request merged**